tests/functional/glusterd/test_gluster_operations_after_removing_firewall.py

```diff
@@ -0,0 +1,241 @@
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+""" Description:
+      Test to check gluster operations after removing firewall
+"""
+
+from glusto.core import Glusto as g
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass
+from glustolibs.gluster.volume_ops import (volume_create, volume_start,
+                                           get_volume_list)
+from glustolibs.gluster.volume_libs import (cleanup_volume)
+from glustolibs.gluster.peer_ops import (peer_probe, peer_detach,
+                                         peer_probe_servers,
+                                         nodes_from_pool_list,
+                                         wait_for_peers_to_connect,
+                                         is_peer_connected)
+from glustolibs.gluster.lib_utils import (form_bricks_list,
+                                          add_services_to_firewall,
+                                          remove_service_from_firewall)
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.mount_ops import mount_volume
+
+
+class TestGlusterOperationsAfterRemovingFirewall(GlusterBaseClass):
+
+    def setUp(self):
+
+        # Adding services list here, so that it can be
+        # used in teardown if setup fails
+        self.services_lst = ['glusterfs', 'nfs', 'rpc-bind']
+
+        # Performing peer detach
+        for server in self.servers[1:]:
+            ret, _, _ = peer_detach(self.mnode, server)
+            if ret != 0:
+                raise ExecutionError("Peer detach failed")
+        self.get_super_method(self, 'setUp')()
+
+    def tearDown(self):
+        # Reset firewall services to the zone
+        if not self.firewall_added:
+            ret = add_services_to_firewall(self.servers[:2],
+                                           self.services_lst, True)
+            if not ret:
+                raise ExecutionError("Failed to add firewall services")
+
+        # Reload firewall services
+        ret = self._reload_firewall_service(self.servers[:2])
+        if not ret:
+            raise ExecutionError("Failed to reload firewall services")
+
+        # Cleanup the volumes
+        vol_list = get_volume_list(self.mnode)
+        if vol_list is not None:
+            for volume in vol_list:
+                ret = cleanup_volume(self.mnode, volume)
```

Comment:
`cleanup_volume()` mostly stop and deletes the volume and does rm -rf on bricks. In this testcase I see that we are mounting the volume as well. I would suggest to use a if-else logic here as shown below:
```
if self.mounted:
   ret = self.umount_volume_and_cleanup_volume([<Servers Where mounted>])
else:
     ret = cleanup_volume(self.mnode, volume)
```

Reply:
Ack